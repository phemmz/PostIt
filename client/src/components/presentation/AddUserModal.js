import React, { Component } from 'react';
import ReactDOM from 'react-dom';
/**
 * AddUserModal class
 */
class AddUserModal extends Component {
	render() {
		return (
			<div id="addUser" className="modal">
        <div className="modal-content">
          <h4 className="green-text text-darken-4">Add User to a Group</h4>
          <div className="row">
            <div className="input-field col s12">
              <select ref="dropdown" id="username" value={this.props.value} onChange={this.props.onChange} style={{display: 'block'}}>
                <option value="" disabled>Add User</option>
                {/* Populate the select dropdown with registered users  */}
                {this.props.appUsers}
              </select> 
            </div>
          </div>
        </div>
        <div className="modal-footer">
          <a className="modal-action modal-close waves-effect waves-green btn-flat">Cancel</a>
          <a onClick={this.props.onClick} className="modal-action modal-close waves-effect waves-green btn-flat">Add User</a>
        </div>
      </div>
		)
	}
}
export default AddUserModal;