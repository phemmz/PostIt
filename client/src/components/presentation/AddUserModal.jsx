import React from 'react';
import PropTypes from 'prop-types';

const AddUserModal = ({
  value, onChange, appUsers, onClick
}) => {
  return (
    <div id="addUser" className="modal">
      <div className="modal-content">
        <h4 className="green-text text-darken-4">Add User to a Group</h4>
        <div className="row">
          <div className="input-field col s12">
            <select id="username" value={value} onChange={onChange} style={{ display: 'block' }}>
              <option value="" disabled>Add User</option>
              {appUsers}
            </select>
          </div>
        </div>
      </div>
      <div className="modal-footer">
        <a className="modal-action modal-close waves-effect waves-green btn-flat">Cancel</a>
        <a
          onClick={onClick}
          className="modal-action modal-close waves-effect waves-green btn-flat"
          role="link"
          tabIndex={0}
        >Add User</a>
      </div>
    </div>
  );
};

AddUserModal.propTypes = {
  value: PropTypes.string.isRequired,
  appUsers: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired
};

export default AddUserModal;
