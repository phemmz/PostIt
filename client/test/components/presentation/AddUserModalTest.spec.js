import React from 'react';
import { mount } from 'enzyme';
import AddUserModal from '../../../src/components/presentation/AddUserModal';

describe('AddUserModal Component', () => {
  let value = 'phemmz';
  const updateUser = () => {
    value = 'new phemmz';
  };
  const appUsers = ['phemmz', 'nawtuaw'];
  const addUser = () => {
    appUsers.push('Nanda');
  };
  it('displays modal for adding user to a group', () => {
    const wrapper = mount(
      <AddUserModal
        value={value}
        onChange={updateUser}
        appUsers={appUsers}
        onClick={addUser}
      />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('value')).toBe(value);
    expect(wrapper.prop('appUsers')).toBe(appUsers);
    expect(wrapper.find('div').exists()).toBe(true);
    expect(wrapper.find('h4').exists()).toBe(true);
    expect(wrapper.find('a').exists()).toBe(true);
    expect(wrapper.find('div').length).toBe(5);
    expect(wrapper.find('h4').length).toBe(1);
    expect(wrapper.find('a').length).toBe(2);
  });
});
