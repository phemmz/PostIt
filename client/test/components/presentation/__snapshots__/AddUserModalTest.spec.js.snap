// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUserModal Component displays modal for adding user to a group 1`] = `
<AddUserModal
  appUsers={
    Array [
      "phemmz",
      "nawtuaw",
    ]
  }
  onChange={[Function]}
  onClick={[Function]}
  value="phemmz"
>
  <div
    className="modal"
    id="addUser"
  >
    <div
      className="modal-content"
    >
      <h4
        className="green-text text-darken-4"
      >
        Add User to a Group
      </h4>
      <div
        className="row"
      >
        <div
          className="input-field col s12"
        >
          <select
            id="username"
            onChange={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
            value="phemmz"
          >
            <option
              disabled={true}
              value=""
            >
              Add User
            </option>
            phemmz
            nawtuaw
          </select>
        </div>
      </div>
    </div>
    <div
      className="modal-footer"
    >
      <a
        className="modal-action modal-close waves-effect waves-green btn-flat"
      >
        Cancel
      </a>
      <a
        className="modal-action modal-close waves-effect waves-green btn-flat"
        onClick={[Function]}
        role="link"
        tabIndex={0}
      >
        Add User
      </a>
    </div>
  </div>
</AddUserModal>
`;
